cross-chain/solana/solana-tbtc-anchor/programs/solana-tbtc-anchor/src/lib.rs

```diff
@@ -0,0 +1,338 @@
+        tbtc.paused = false;
+        Ok(())
+    }
+
+    pub fn add_minter(ctx: Context<AddMinter>) -> Result<()> {
+        let minter_info = &mut ctx.accounts.minter_info;
+        minter_info.minter = ctx.accounts.minter.key();
+        minter_info.bump = *ctx.bumps.get("minter_info").unwrap();
+
+        ctx.accounts.tbtc.minters += 1;
+        Ok(())
+    }
+
+    pub fn remove_minter(ctx: Context<RemoveMinter>, _minter: Pubkey) -> Result<()> {
+        require!(ctx.accounts.tbtc.minters > 0, TbtcError::NoMinters);
+        ctx.accounts.tbtc.minters -= 1;
+        Ok(())
+    }
+
+    pub fn add_guardian(ctx: Context<AddGuardian>) -> Result<()> {
+        let guardian_info = &mut ctx.accounts.guardian_info;
+        guardian_info.guardian = ctx.accounts.guardian.key();
+        guardian_info.bump = *ctx.bumps.get("guardian_info").unwrap();
+
+        ctx.accounts.tbtc.guardians += 1;
+        Ok(())
+    }
+
+    pub fn remove_guardian(ctx: Context<RemoveGuardian>, _minter: Pubkey) -> Result<()> {
+        require!(ctx.accounts.tbtc.guardians > 0, TbtcError::NoGuardians);
+        ctx.accounts.tbtc.guardians -= 1;
+        Ok(())
+    }
+
+    pub fn setup_mint(_ctx: Context<SetupMint>) -> Result<()> {
+        msg!("setup mint complete\n\n\n");
+        Ok(())
+    }
+
+    pub fn mint(ctx: Context<Mint>, amount: u64) -> Result<()> {
+        msg!("start mint\n\n\n");
+        let seed = b"tbtc-mint";
+        let key_seed = ctx.accounts.tbtc.key();
+        let mint_bump = ctx.accounts.tbtc.token_bump;
+        let signer: &[&[&[u8]]] = &[&[seed, key_seed.as_ref(), &[mint_bump]]];
+
+        let cpi_ctx = CpiContext::new_with_signer(
+            ctx.accounts.token_program.to_account_info(),
+            SplMintTo {
+                mint: ctx.accounts.tbtc_mint.to_account_info(),
+                to: ctx.accounts.recipient_account.to_account_info(),
+                authority: ctx.accounts.tbtc_mint.to_account_info(),
+            },
+            signer,
+        );
+        mint_to(cpi_ctx, amount)?;
+
+        Ok(())
+    }
+
+    pub fn pause(ctx: Context<Pause>) -> Result<()> {
+        ctx.accounts.tbtc.paused = true;
+        Ok(())
+    }
+
+    pub fn unpause(ctx: Context<Unpause>) -> Result<()> {
+        ctx.accounts.tbtc.paused = false;
+        Ok(())
+    }
+}
+
+#[account]
+#[derive(Default)]
+pub struct Tbtc {
+    creator: Pubkey,
+    token_mint: Pubkey,
+    token_bump: u8,
+    minters: u8,
+    guardians: u8,
+    paused: bool,
+}
+
+#[account]
+#[derive(Default)]
+pub struct MinterInfo {
+    minter: Pubkey,
+    bump: u8,
+}
+
+#[account]
+#[derive(Default)]
+pub struct GuardianInfo {
+    guardian: Pubkey,
+    bump: u8,
+}
+
+#[derive(Accounts)]
+pub struct Initialize<'info> {
+    // Use PDA for the mint address
+    // so we can sign for it from the program
+    #[account(
+        init,
+        seeds = [b"tbtc-mint", tbtc.key().as_ref()],
+        bump,
+        payer = creator,
+        mint::decimals = 8,
```

Comment:
I just chose a number matching Bitcoin for the initial testing because I didn't pay attention to the exact decimal count of SPL tokens when I wrote that part. There is no reason to keep using 8 decimals instead of the standard 9.